x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -865,6 +872,24 @@ function getCorrespondingAction(
   }
 }
 
+function removeInternalTags(
+  doc: SavedObjectUnsanitizedDoc<RawRule>
+): SavedObjectUnsanitizedDoc<RawRule> {
+  const {
```

Comment:
Should this only be applied to security solutions rules?

Reply:
@vitaliidm You would have to test that `doc.attributes.alertTypeId` is one of the security rule types. I believe there is already a function in this file that would do this: `isDetectionEngineAADRuleType`

Reply:
thanks @ymao1, I've added rule type check as advised